ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/P2PDumpManager.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.util;
+
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+
+public class P2PDumpManager {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final String GOSSIP_DECODING_ERROR_DIR = "gossip_decoding_error_messages";
+  private static final String GOSSIP_REJECTED_DIR = "rejected_gossip_messages";
+  private static final String INVALID_BLOCK_DIR = "invalid_blocks";
+
+  private final Path directory;
+
+  public P2PDumpManager(final Path directory) {
+    this.directory = directory;
+    if (this.directory.resolve(GOSSIP_DECODING_ERROR_DIR).toFile().mkdir()) {
```

Comment:
We should use `mkdirs()` so it will also handle any intermediate folders that do not exist.